x/claim/keeper/mission.go

```diff
@@ -57,29 +49,26 @@ func (k Keeper) GetAllMission(ctx sdk.Context) (list []types.Mission) {
 	return
 }
 
-// CompleteMission triggers the completion of the mission and distribute the claimable portion of airdrop to the user
-// the method fails if the mission has already been completed
-func (k Keeper) CompleteMission(ctx sdk.Context, missionID uint64, address string) error {
-	airdropSupply, found := k.GetAirdropSupply(ctx)
-	if !found {
-		return errors.Wrapf(types.ErrAirdropSupplyNotFound, "airdrop supply is not defined")
-	}
-
+// CompleteMission saves the completion of the mission
+func (k Keeper) CompleteMission(
+	ctx sdk.Context,
+	missionID uint64,
+	address string,
+) (claimed math.Int, err error) {
```

Comment:
We should maybe also return a boolean `isClaimed` as returning `claimed` alone can be misleading since this value not always set

Reply:
From what I can see, `claimed` is never set actually... What does this value represent?

Reply:
IMO `isClaimed` can be redundant because we can see the value `claimed` if it is zero, nothing was claimed, or the opposite

Reply:
> From what I can see, `claimed` is never set actually... What does this value represent?

if the `AirdropStart` time reaches, the `CompleteMission` will claim automatically, if not, the user needs to claim manually

Reply:
> > From what I can see, `claimed` is never set actually... What does this value represent?
> 
> if the `AirdropStart` time reaches, the `CompleteMission` will claim automatically, if not, the user needs to claim manually

Thanks I deleted my comment because I saw after than `claimed` can be set from the call to `ClaimMission` :pray:

Reply:
> IMO isClaimed can be redundant because we can see the value claimed if it is zero, nothing was claimed, or the opposite

OK, it should be stated in the comment then

Reply:
@Pantani can we precise in the method doc that `CompleteMission` claims it if `AirdropStart` is already reached and `claimed` is 0 if the mission is not claimed

Reply:
I added here:

https://github.com/ignite/modules/pull/77/files#diff-3a7e98e2e3b5aeb6fd5e3933637c2a48179ef5c0f82a8b264bf61ffb602d6d8aR35-R40

Reply:
I meant for the `CompleteMission` as well to describe the method signature